### PR TITLE
Improve checks and tests for preprocess.py

### PR DESCRIPTION
**Pull request by bw4sz**
This is a follow to PR #552, we needed a test, and to warn users if rasters were in Channel x height x width before trying to coerce them to height x width x channel. I also changed base_dir to save_dir based on the deprecation warning and changed to use tmpdir when we aren't check the output of ```preprocess.split_raster```.

### CODE REVIEW

**Comment by bw4sz** [comment]
Not sure why the tag push is this PR?

**Comment by bw4sz** [comment]
This isn't quite right yet, I was testing this on hpg,

```
(base) [b.weinstein@c39a-s5 MillionTrees]$ conda activate DeepTreeAttention
(DeepTreeAttention) [b.weinstein@c39a-s5 MillionTrees]$  cd /home/b.weinstein/MillionTrees ; /usr/bin/env /orange/ewhite/b.weinstein/miniconda3/envs/DeepTreeAttention/bin/python /home/b.weinstein/.vscode-server/extensions/ms-python.python-2023.20.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 44399 -- /home/b.weinstein/MillionTrees/data_prep/create_train_test.py 
/orange/ewhite/b.weinstein/miniconda3/envs/DeepTreeAttention/lib/python3.11/site-packages/deepforest/preprocess.py:192: UserWarning: Input rasterio had shape (40665, 84370, 4), assuming channels first. Converting to channels last
  warnings.warn(
/orange/ewhite/b.weinstein/miniconda3/envs/DeepTreeAttention/lib/python3.11/site-packages/deepforest/preprocess.py:200: UserWarning: Input rasterio had non-3 band shape of (84370, 4, 40665), ignoring alpha channel
  warnings.warn("Input rasterio had non-3 band shape of {}, ignoring "
/orange/ewhite/b.weinstein/miniconda3/envs/DeepTreeAttention/lib/python3.11/site-packages/deepforest/preprocess.py:203: RuntimeWarning: invalid value encountered in cast
  numpy_image = numpy_image[:, :, :3].astype("uint8")
```

**Comment by bw4sz** [comment]
Okay, @henrykironde should be ready.

File: tests/test_preprocess.py [hunk 1]
```diff
@@ -168,11 +168,35 @@ def test_split_raster_empty(config):
...
\ No newline at end of file
+
+def test_split_raster_4_band_warns(config, tmpdir):
+    # Confirm that the rasterio channel order is C x H x W
+    assert rasterio.open(get_data("OSBS_029.tif")).read().shape[0] == 3
+
+    # Create a 4 band image in the wrong channel order, it should be H x W x C
+    numpy_image = np.zeros((4, 400, 400), dtype=np.uint8)
+    
+    with pytest.warns(UserWarning):
```
Comment (henrykironde) [inline tests/test_preprocess.py:186]: @ Ben Why are we repeating the procedure ?

**Review by henrykironde — COMMENTED** [review]
@bw4sz this looks good, I have made a comment on the tests.

The version bump is from the settings we gave the package in [setup.cfg](https://github.com/weecology/DeepForest/blob/main/setup.cfg)

**Comment by bw4sz** [comment]
> @bw4sz this looks good, I have made a comment on the tests.
> 
> The version bump is from the settings we gave the package in [setup.cfg](https://github.com/weecology/DeepForest/blob/main/setup.cfg)

Yes, but we already pushed that. 1.3.1 is already up on pypi, 

https://pypi.org/project/deepforest/

image

File: tests/test_preprocess.py (hunk 1 above)
Comment (bw4sz) [reply tests/test_preprocess.py:186]: Once for when the channel order channels x height x width and once for when the channel order is height x width x channels. The function should work and warn users in either situation. Before it was assuming the correct shape, which is channel x height x width, without actually checking it was true.

**Comment by bw4sz** [comment]
@henrykironde I could use this next week, let me if there are concerns here.

**Review by henrykironde — APPROVED** [review]

**Pull request merged**